File: Medium/Minimum times A has to be repeated such that B is a substring of it/minimum-times-a-has-to-be-repeated-such-that-b-is-a-substring-of-it.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
//User function Template for C++

class Solution {
  public:
    int minRepeats(string A, string B) {
      if(A==B) return 1;
        if(A.length()>B.length()) return -1;
        string dup=A;
        int count=1;
        while(A.length()<B.length()){
            A+=dup;
            count++;
        }
        if(A.find(B)!=-1) return count;
        A+=dup;
        count++;
        if(A.find(B)!=-1) return count;
        return -1;
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    scanf("%d ",&t);
    while (t--) {
        string A,B;
        getline(cin,A);
        getline(cin,B);

        Solution ob;
        cout << ob.minRepeats(A,B) << endl;
    }
    return 0;
}
// } Driver Code Ends

File: Medium/Minimum times A has to be repeated such that B is a substring of it/minimum-times-a-has-to-be-repeated-such-that-b-is-a-substring-of-it.java
//{ Driver Code Starts
import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            String A = read.readLine();
            String B = read.readLine();

            Solution ob = new Solution();
            System.out.println(ob.minRepeats(A,B));
        }
    }
}
// } Driver Code Ends


//User function Template for Java

class Solution {
    static int minRepeats(String A, String B) {
        int n = B.length(), m = A.length();
        
        int max = n/m+1;
        int min = 1;
        String S = A;
        
        for(int i=0; i<=max; i++)
        {
            if(A.contains(B))
            {
                return min;
                
            }
            
            A = A + S;
            min++;
            
        }
        
        return -1;
    }
};